src/layouts/MainLayout.js

```diff
@@ -46,15 +46,18 @@ const MainLayout = ({ children }) => {
     ? null
     : `${siteMetadata.repository}/blob/main/${fileRelativePath}`;
 
-  useEffect(() => {
-    const consentValue = Cookies.get(gdprConsentCookieName) === 'true';
-    consentValue && setCookieConsent(true);
-  }, []);
-
   useEffect(() => {
     setIsMobileNavOpen(false);
   }, [location.pathname]);
 
+  useEffect(() => {
+    if (cookieConsent && previousCookieConsent === false) {
```

Comment:
If the user has accepted cookies, does it automatically start tracking without needing to call `window.trackGoogleAnalytics`?

Reply:
yes! so it runs on route change, but if the user has already accepted the cookies on a previous visit then it doesn't need to call `window.trackGoogleAnalytics`

Reply:
Just to also clarify, if you reload a page after accepting cookies, does it record that as a hit without needing to change routes first?

Reply:
yes! it does record as a hit without needing to change routes